src/library/controllers/message/interface.js

```diff
@@ -58,146 +58,168 @@ export default (options = {}) => ({
+                                }
 
-                        // Merge new styles into previous styles
-                        // Especially useful when combining inline attribute styles with JS API styles
-                        if (state.style && messageProps.style) {
-                            const totalStyle = objectMerge(state.style, messageProps.style);
-                            state.style = totalStyle;
-                            messageProps.style = totalStyle;
-                        }
+                                // Filter out undefined to prevent overwriting previous values
+                                const updatedMessageProps = objectEntries(messageProps).reduce(
+                                    (acc, [key, val]) =>
+                                        typeof val === 'undefined' ? acc : Object.assign(acc, { [key]: val }),
+                                    {}
+                                );
+
+                                return updateProps(updatedMessageProps)
+                                    .then(() => globalEvent.trigger('render'))
+                                    .then(resolve);
+                            } catch (err) {
+                                // We only want console.warn to be called once
+                                // check for the known error offer_validation_error and if it’s not that error then we should throw the error again so it gets surfaced
+                                //  catch that known/safe error, but allow any other errors to escape through
+                                if (err.message === 'offer_validation_error') {
+                                    return resolve();
+                                }
+                                return reject(err);
+                            }
+                        };
 
-                        // Filter out undefined to prevent overwriting previous values
-                        const updatedMessageProps = objectEntries(messageProps).reduce(
-                            (acc, [key, val]) =>
-                                typeof val === 'undefined' ? acc : Object.assign(acc, { [key]: val }),
-                            {}
-                        );
-
-                        return updateProps(updatedMessageProps).then(() => globalEvent.trigger('render'));
-                    } catch (err) {
-                        // We only want console.warn to be called once
-                        // check for the known error offer_validation_error and if it’s not that error then we should throw the error again so it gets surfaced
-                        //  catch that known/safe error, but allow any other errors to escape through
-                        if (err.message === 'offer_validation_error') {
-                            return undefined;
+                        if (mapIndex === 0) {
+                            return renderOrUpdateMessage();
```

Comment:
We can remove the `consistent-return` eslint warning by updating this slightly
```suggestion
                            renderOrUpdateMessage();
                            return;
```